gubernator/main.py

```diff
@@ -0,0 +1,198 @@
+#!/usr/bin/env python
+#
+# Copyright 2016 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import datetime
+import functools
+import hashlib
+import json
+import logging
+import re
+import os
+import urllib
+import zlib
+
+import webapp2
+import jinja2
+
+from google.appengine.api import memcache
+
+import lib.defusedxml.ElementTree as ET
+import lib.cloudstorage as gcs
+
+BUCKET_WHITELIST = {
+    'kubernetes-jenkins',
+}
+
+GITHUB_VIEW_TEMPLATE = 'https://github.com/kubernetes/kubernetes/blob/%s/%s#L%s'
+
+JINJA_ENVIRONMENT = jinja2.Environment(
+    loader=jinja2.FileSystemLoader(os.path.dirname(__file__) + '/templates'),
+    extensions=['jinja2.ext.autoescape'],
+    trim_blocks=True,
+    autoescape=True)
+
+
+def jinja_filter(func):
+    JINJA_ENVIRONMENT.filters[func.__name__.replace('format_', '')] = func
+    return func
+
+
+@jinja_filter
+def format_timestamp(ts):
```

Comment:
Please use longer values for function variables (func and stamp are fine but ts for timestamp is too short, same with inp which I don't even know what it is supposed to be (inpath?)